api/src/main/java/com/datastrato/gravitino/rel/SupportsPartitions.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel;
+
+import com.datastrato.gravitino.exceptions.NoSuchPartitionException;
+import com.datastrato.gravitino.exceptions.PartitionAlreadyExistsException;
+import com.datastrato.gravitino.rel.expressions.partitions.Partition;
+import java.util.Map;
+
+public interface SupportsPartitions extends Table {
+
+  /**
+   * List all partition names of the table
+   *
+   * @return The list of partition names
+   */
+  String[] listPartitionNames();
+
+  /**
+   * List all partitions
+   *
+   * @return The list of partitions
+   */
+  Partition[] listPartitions();
+
+  /**
+   * get a partition by partition name
+   *
+   * @param partitionName the name of the partition
+   * @return the partition
+   * @throws NoSuchPartitionException if the partition does not exist
+   */
+  Partition getPartition(String partitionName) throws NoSuchPartitionException;
+
+  /**
+   * Check if a partition exists.
+   *
+   * @param partitionName The name of the partition.
+   * @return True if the partition exists, false otherwise.
+   */
+  default boolean partitionExists(String partitionName) {
+    try {
+      getPartition(partitionName);
+      return true;
+    } catch (NoSuchPartitionException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Add a partition with specified name and properties to the table.
+   *
+   * @param partitionName The identifier of the partition.
+   * @param properties The properties of the partition.
+   * @return The created partition.
+   * @throws PartitionAlreadyExistsException If the partition already exists.
+   */
+  Partition addPartition(String partitionName, Map<String, String> properties)
+      throws PartitionAlreadyExistsException;
+
+  /**
+   * Alter a partition with specified identifier.
+   *
+   * @param partitionName The identifier of the partition.
+   * @param changes The changes to apply.
+   * @return The altered partition.
+   * @throws NoSuchPartitionException If the partition does not exist.
+   */
+  Partition alterPartition(String partitionName, PartitionChange... changes)
```

Comment:
PartitionChange supports rename for now, Is there any scence to support change partition properties like spark?

Reply:
Not yet, but it can be added later if there is a similar requirement.